kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -327,26 +397,27 @@ public static boolean delete(URI uri) {
   }
 
   /**
-   * Delete a {@link Dataset} identified by the given dataset URI string.
+   * Delete a {@link Dataset} identified by the given dataset URI string. Kite takes care
+   * of any housekeeping, including deleting data and any metadata stored separately.
    * <p>
-   * URI formats are defined by {@code Dataset} implementations, but must begin
-   * with "dataset:".
+   * URIs must begin with {@code dataset:}. The remainder of
+   * the URI is implementation specfic, depending on the dataset scheme.
    *
-   * @param uri a {@code Dataset} URI string.
-   * @return {@code true} if any data or metadata was removed, or {@code false}
+   * @param uri a {@code Dataset} URI string
+   * @return {@code true} if any data or metadata is removed, or {@code false}
    */
   public static boolean delete(String uri) {
     return delete(URI.create(uri));
   }
 
   /**
-   * Check if a {@link Dataset} identified by the given URI exists.
+   * Check whether a {@link Dataset} identified by the given URI exists.
    * <p>
-   * URI formats are defined by {@code Dataset} implementations, but must begin
-   * with "dataset:".
+   * URIs must begin with {@code dataset:}. The remainder of
+   * the URI is implementation specfic, depending on the dataset scheme.
    *
-   * @param uri a {@code Dataset} URI.
-   * @return {@code true} if the dataset exists, {@code false} otherwise
+   * @param uri a {@code Dataset} URI
+   * @return {@code true} if the dataset exists, or {@code false}
```

Comment:
I like for the wording to be more clear. To avoid the problem another way, we can use "{@code true} if and only if the dataset exists."